app/assets/javascripts/host_edit.js

```diff
@@ -295,10 +295,7 @@ function serializeForm() {
 
 //This function decides for a given input element is it set explicitly by the user.
 function isExplicit(index, element) {
-  if (element.nextSibling == undefined) return true;
-  if (element.nextSibling.children == undefined) return true;
-  if (element.nextSibling.children[0] == undefined) return true;
-  if (element.nextSibling.children[0].dataset == undefined) return true;
+  if (!element.classList.contains("has-explicit")) return true;
   return element.nextSibling.children[0].dataset.explicit == 'true';
```

Comment:
I was thinking along these lines:

<pre>
button = element.nextAll().find('button.has-explicit');  // find associated "inherit" button
return if (button.length == 0);
return button[0].dataset.explicit == 'true';
</pre>

But there's nothing much in it.  Using a selector may just tidy this line up to address @tbrisker's original comment here, whether or not the class stays on the select or the button.